instrumentation/camel-2.20/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SqsCamelTest.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachecamel.aws;
+
+import com.google.common.collect.ImmutableMap;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class SqsCamelTest {
+
+  @RegisterExtension
+  public static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static final AwsConnector awsConnector = AwsConnector.elasticMq();
+
+  @AfterAll
+  static void cleanUp() {
+    awsConnector.disconnect();
+  }
+
+  private static void waitAndClearSetupTraces(String queueUrl, String queueName) {
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> AwsSpanAssertions.sqs(span, 0, "SQS.CreateQueue", queueUrl, queueName)));
+    testing.clearData();
+  }
+
+  @Test
+  void camelSqsProducerToCamelSqsConsumer() {
+    String queueName = "sqsCamelTest";
+    String queueUrl = awsConnector.createQueue(queueName);
+    waitAndClearSetupTraces(queueUrl, queueName);
+
+    CamelSpringApplication camelApp =
+        new CamelSpringApplication(
+            awsConnector, SqsConfig.class, ImmutableMap.of("queueName", queueName));
+
+    camelApp.start();
+    camelApp.producerTemplate().sendBody("direct:input", "{\"type\": \"hello\"}");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace
+                .hasSize(1)
+                .hasSpansSatisfyingExactly(
+                    span -> AwsSpanAssertions.sqs(span, 0, "SQS.ListQueues")),
+        trace ->
+            trace
+                .hasSize(5)
```

Comment:
`hasSize()` is not really necessary, since `hasSpansSatisfyingExactly()` already verifies the size
```suggestion
```